### DISCUSSION TITLE
Database cleanup

### DISCUSSION THREAD

**Discussion by rizlas** [healthchecks/healthchecks#628] [Q&A]
Hi,
we are currently running a self hosted docker HC. We always ran it as docker even without the official image. Currently we spotted that api_ping table contains 10kk rows. How database cleanup works? (using mysql)

> Healthchecks deletes old entries from api_ping and api_notification tables automatically. By default, Healthchecks keeps the 100 most recent pings for every check. You can set the limit higher to keep a longer history: go to the Administration Panel, look up user's Profile and modify its "Ping log limit" field.

Thank you

**Comment by SuperSandro2000** [comment]
https://github.com/healthchecks/healthchecks#database-cleanup

**Reply by rizlas** [reply]
Thanks Sandro, but this is exactly what I already quoted and it is not very useful.

**Comment by cuu508** [comment]
What version of Healthchecks are you running? Versions starting from v1.24.0 clean up the `api_ping` table automatically.

You can check the version by going to Django admin (`/admin/`), it will be displayed in the page header.

**Reply by rizlas** [reply]
Currently running v2.0.1

**Reply by cuu508** [reply]
Then you're all set. 
The cleanup logic is here – https://github.com/healthchecks/healthchecks/blob/01be572a5427b8ce6bb3f800684a0ffa11739e05/hc/api/models.py#L336

**Reply by rizlas** [reply]
Thank you, we migrated to postgres and pruning seems to works fine now.

**Reply by cuu508** [reply]
Do you think the migration to Postgres fixed it? I expect the pruning to work with MySQL and SQLite too. If it doesn't, that's a bug.

**Reply by rizlas** [reply]
I'm not confident to say that is a bug. Because we tried MySql with a clean install and mocking pings. Pruning was working fine.
In our legacy mysql with 200 checks we had 10kk pings (so pruning was not working). We manually pruned them with django shell and migrated the entire db to postgres.

**Reply by cuu508** [reply]
The default ping log limit is 100 entries. 200 checks X 100 entries would be 20'000 entries total. 
By 10kk do you mean 10 kilo kilo, i.e., 10'000'000?

**Reply by rizlas** [reply]
With 10kk i mean 10.000.000 rows yes.

**Reply by cuu508** [reply]
And with the legacy mysql, were you also running v1.24.0 or later?

**Reply by rizlas** [reply]
In December we switched to official Docker image 1.25. Last week we upgraded to 2.0.1

**Reply by cuu508** [reply]
Before upgrading to 1.25, were you running `manage.py prunepings` regularly?

**Reply by rizlas** [reply]
Cant say that, I'm a new maintainer but I guess the answer is no.

**Reply by cuu508** [reply]
That explains it then. Before upgrading to v1.25, the pings were accumulating because the `manage.py prunepings` command was not being run.

After upgrading v1.25 the automatic pruning kicks in, but with a couple caveats:

* pings are pruned infrequently, and only for checks that receive new pings. After upgrading v1.25 it would take time for the number of rows in `api_ping` to gradually decrease
* even after the number of rows has gone down, the database may not reclaim disk space due to fragmentation. In Postgres that's fixed by `VACUUM FULL`, perhaps something similar also needs to be done on MySQL.

**Reply by rizlas** [reply]
Yeah it looks like what happened. We also switched to postgres because imho is more reliable. Anyway really appreciate your explanation and support. Thank you!

**Reply by cuu508** [reply]
I accept tokens of appreciation here – https://github.com/sponsors/healthchecks/ :-)
